src/main/java/com/walmartlabs/x12/asn856/DefaultAsn856Parser.java

```diff
@@ -14,39 +28,19 @@
  * delivered.
  *
  */
-public class DefaultAsn856Parser implements X12Parser<Asn856> {
+public class DefaultAsn856Parser extends AbstractStandardX12Parser<Asn856> {
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAsn856Parser.class);
 
-    /**
-     * parse the ASN 856 transmission into a representative Java object
-     *
-     * @return {@link Asn856}
-     * @throws X12ParserException
-     */
     @Override
-    public Asn856 parse(String sourceData) {
-        Asn856 asn856 = null;
-
-        if (!StringUtils.isEmpty(sourceData)) {
-            asn856 = new Asn856();
-            List<X12Segment> segmentLines = this.splitSourceDataIntoSegments(sourceData);
-            if (!this.isValidEnvelope(segmentLines)) {
-                throw new X12ParserException("invalid envelope");
-            } else {
-                // TODO: implement the parser
-            }
-        }
-
-        return asn856;
+    protected AbstractStandardX12Document createX12Document() {
+        return new Asn856();
     }
 
-    /**
-     * check for valid envelope (header and trailer segments)
-     * @param segmentLines
-     * @return
-     */
-    protected boolean isValidEnvelope(List<X12Segment> segmentLines) {
-        // TODO: determine valid envelope
-        return true;
+    @Override
+    protected void parseCustom(List<X12Segment> segmentLines, AbstractStandardX12Document x12Doc) {
```

Comment:
`parseOtherDetails`?

Reply:
still thinking about best way to handle custom segments